ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -393,35 +394,22 @@ private void dumpFunctionMetadata(String dbName, Path dumpRoot) throws SemanticE
     }
   }
 
-  /**
-   *
-   * @param ast
-   * @param dbName
-   * @param tblName
-   * @param dbRoot
-   * @return tbl dumped path
-   * @throws SemanticException
-   */
-  private Path dumpTbl(ASTNode ast, String dbName, String tblName, Path dbRoot) throws SemanticException {
-    Path tableRoot = new Path(dbRoot, tblName);
+  private void dumpTable(ASTNode ast, String dbName, String tblName, Path dbRoot)
+      throws SemanticException {
     try {
-      URI toURI = EximUtil.getValidatedURI(conf, tableRoot.toUri().toString());
       TableSpec ts = new TableSpec(db, conf, dbName + "." + tblName, null);
-
-      ExportSemanticAnalyzer.prepareExport(ast, toURI, ts, getNewReplicationSpec(), db, conf, ctx,
-          rootTasks, inputs, outputs, LOG);
+      TableExport.Paths exportPaths = new TableExport.Paths(ast, dbRoot, tblName, conf);
```

Comment:
Well i dont call myself an expert, but I think object design tells you to create classes such that each one of them do a limited set of tasks, makes it easy to understand the code. The idea with Paths class is that it provides behavior's that allow to find what path what kind of data has to be exported, along with making sure that respective conditions are met to export data at the said path, For Ex: create the directories, if not already present for exporting data. This also provides one class to look up how different paths are created to export _metadata as well data for table/partition without looking at specific places, Now in your latest argument you have deemed the existence of Paths as not required which is disjoint to  point of the object creation.  

As you said there is nothing immediately that is visible for different implementations for this, but does this warrant that we dont create the new class? for ex there are no multiple implementations  of org.apache.hadoop.hive.ql.Context class. Furthermore, for the cloud use case with different file systems, i am not sure if we have to create the directories or objects differently. 

Table export is written to just take care of writing metadata / data out based on certain conditions, It is using a helper Paths class to determine where to write the data.